fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/DateTimeArithmetic.java

```diff
@@ -37,315 +38,543 @@ public class DateTimeArithmetic {
      * datetime arithmetic function date-add.
      */
     @ExecFunction(name = "date_add", argTypes = {"DATE", "INT"}, returnType = "DATE")
-    public static Expression dateAdd(DateLiteral date, IntegerLiteral day) {
-        return daysAdd(date, day);
+    public static Expression dateAdd(DateLiteral date, IntegerLiteral day) throws Exception {
+        Expression res = daysAdd(date, day);
+        if (res instanceof NullLiteral) {
```

Comment:
remove these code, we should not depend on nullable to check if the value is out of range